ush/cdo_post_ocean_format.sh

```diff
@@ -0,0 +1,193 @@
+#
+#   Arguments:
+#
+#     variable_file: ASCII formatted file containing netCDF variables
+#                      and the respective metadata attributes to be
+#                      updated/added; the supported format is as
+#                      follows.
+#
+#     <netCDF variable name> <netCDF metadata attribute name>
+#       <netCDF metadata values>
+#
+#       An example using the format described above is as follows.
+#
+#     SST coordinates geolon,geolat,time
+#     uo coordinates geolon_u,geolon_u,time
+#     vo coordinates geolon_v,geolon_v,time
+#
+#       The example above will perform the following task using this
+#       script.
+#
+#       * Assign the netCDF metadata attribute `coordinates` for
+#         variable `SST` the values `geolon,geolat,time` and update
+#         the `output_netcdf` file path.
+#
+#       * Assign the netCDF metadata attribute `coordinates` for
+#         variable `uo` the values `geolon_u,geolat_u,time` and update
+#         the `output_netcdf` file path.
+#
+#       * Assign the netCDF metadata attribute `coordinates` for
+#         variable `vo` the values `geolon_v,geolat_v,time` and update
+#         the `output_netcdf` file path.
+#
+#     input_netcdf: The netCDF-formatted file path containing the
+#                     variables defined in `variable_file`.
+#
+#     output_netcdf: A netCDF-formatted file path to contain the
+#                      specified variables remapped to the destination
+#                      grid projection.
+#######
+
+# Collect the command line arguments and check the validity.
+variable_file="${1}"
+input_path="${2}"
+output_path="${3}"
+
+#######
+
+if [[ "$#" -ne 3 ]]; then
+    echo "Usage: $0 <variable_file> <input_path> <output_path>"
+    exit 100
+fi
+
+#######
+
+# _comma_split_string - Split a comma-delimited string into an array.
+#
+# Description:
+#   This function takes a comma-delimited string as input and splits
+#     it into an array. Each element in the resulting array is
+#     obtained by splitting the input string at commas and then
+#     removing leading and trailing spaces.
+#
+# Parameters:
+#   $1 - The comma-delimited string to split.
+#
+# Global Variables:
+#   global_array - An array containing the split elements.
+#
+# Example usage:
+#   _comma_split_string "item1,item2 item3,item4"
+#   for element in "${global_array[@]}"; do
+#       echo "$element"
+#   done
+#
+# This example will split the input string into individual elements
+#   and print each element on a separate line.
+function _comma_split_string() {
+    local string="${1}"
+
+    local local_array=()
+    global_array=()
+    IFS="," read -ra items <<< "${string}"
+    for item in "${items[@]}"; do
+        local_array+=("${item} ")
+    done
+    for item in "${local_array[@]}"; do
+        IFS=" " read -ra items <<< "${item}"
+        for element in "${items[@]}"; do
+            global_array+=("${element} ")
+        done
+    done
+}
```

Comment:
```suggestion
function _comma_split_string() {
    local string="${1}"
    local var_name="${2}"

    # Declare local_array as a reference to the desired array name
    declare -n local_array="${var_name}"
    IFS="," read -ra items <<< "${string}"
    for item in "${items[@]}"; do
        local_array+=("${item}")
    done
}
```

Reply:
Credit to ChatGPT for the `declare -n` trick. I was having trouble using declare to copy the array after-the-fact, but now we don't have to.